lib/Reader/readerir.cpp

```diff
@@ -1412,8 +1412,10 @@ Type *GenIR::getClassType(CORINFO_CLASS_HANDLE ClassHandle, bool IsRefClass,
       CORINFO_FIELD_HANDLE FieldHandle = getFieldInClass(ClassHandle, I);
       if (FieldHandle == nullptr) {
         // Likely a class that derives from System.__ComObject. See
-        // LLILC issue #557.
-        throw NotYetImplementedException("can't account for all fields");
+        // LLILC issue #557. We'll just have to cope with an incomplete
+        // picture of this type.
+        assert(IsRefClass && "need to see all fields of value classes");
+        break;
       }
       ASSERT(FieldHandle != nullptr);
```

Comment:
Nit: this ASSERT is not needed. We really need to make asserts uniform. Here we have ASSERT and assert almost next to each other.